src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -617,6 +617,122 @@ public function testDumpDateTime($dateTime, $expected)
         $this->assertSame($expected, Inline::dump($dateTime));
     }
 
+    /**
+     * @dataProvider getNumericKeyData
+     */
+    public function testDumpNumericKeyAsString(array|int $input, int $flags, string $expected)
+    {
+        $this->assertSame($expected, Inline::dump($input, $flags));
+    }
+
+    public function getNumericKeyData()
+    {
+        yield 'Int with flag' => [
+            200,
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            "'200'",
+        ];
+
+        yield 'Int without flag' => [
+            200,
+            0,
+            '200',
+        ];
```

Comment:
This is tested elsewhere already. Let's focus on cases with `Yaml::DUMP_NUMERIC_KEY_AS_STRING` enabled.